File: examples/fib.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>

long fib(long n) {
    if (n <= 1) {
        return n;
    } else {
        return fib(n - 1) + fib(n - 2);
    }
}

int main(int argc, char* argv[]) {
    int iterations = 1;
    if (argc > 1) {
        iterations = atoi(argv[1]);
    }
    
    struct timespec start, end;
    clock_gettime(CLOCK_MONOTONIC, &start);
    
    long result = 0;
    for (int i = 0; i < iterations; i++) {
        result = fib(35);
    }
    
    clock_gettime(CLOCK_MONOTONIC, &end);
    
    printf("Result: %ld\n", result);
    printf("Iterations: %d\n", iterations);
    printf("Time: %.6f seconds\n", 
           (end.tv_sec - start.tv_sec) + (end.tv_nsec - start.tv_nsec) / 1e9);
    printf("Avg per iteration: %.6f seconds\n",
           ((end.tv_sec - start.tv_sec) + (end.tv_nsec - start.tv_nsec) / 1e9) / iterations);
    
    return 0;
}

File: examples/fib.py
import time
import sys

def fib_helper(n: int, a: int, b: int):
    if n <= 1:
        return a
    else:
        return fib_helper(n - 1, b, a + b)

def fib(n: int):
    return fib_helper(n, 0, 1)

if __name__ == "__main__":
    iterations = 1
    if len(sys.argv) > 1:
        iterations = int(sys.argv[1])
    
    start = time.time()
    
    result = 0
    for i in range(iterations):
        result = fib(35)
    
    end = time.time()
    
    print(f"Result: {result}")
    print(f"Iterations: {iterations}")
    print(f"Time: {end - start:.6f} seconds")
    print(f"Avg per iteration: {(end - start) / iterations:.6f} seconds")
